spec/PeriodicBackgroundSync-index.bs

```diff
@@ -71,57 +71,70 @@ Introduction {#intro}
-<div dfn-for="periodicsync registration">
-A <dfn>service worker registration</dfn>, which is the [=/service worker registration=] associated with the [=context object=] the {{PeriodicSyncManager}} belongs to.
+A <dfn>tag</dfn>, which is a {{DOMString}}.
 
-A <dfn>tag</dfn>, which is a {{DOMString}}. Within one [=list of periodicsync registrations=] each [=periodicsync registration=] MUST have a unique [=periodicsync registration/tag=]. Periodic Background Sync doesn't share namespace with Background Sync, so an [=origin=] can have registrations of both types with the same tag.
+Note: Periodic Background Sync doesn't share namespace with Background Sync, so an [=origin=] can have registrations of both types with the same tag.
 
-<dfn>options</dfn>, which is a dictionary containing [=options/minInterval=].  Enclosing [=options=] in a dictionary allows this spec to be extended with more [=options=] in the future without adversely affecting existing usage.
+<dfn>minimum interval</dfn> (a long long), which is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen. [=minimum interval=] is a suggestion to the user agent.
 
-<div dfn-for=options><dfn>minInterval</dfn> (a long long), which is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen. [=options/minInterval=] is a suggestion to the user agent. The actual interval at which [=periodicSync event|periodicSync events=] are fired MUST be greater than or equal to this.</div>
+Note: The actual interval at which [=periodicsync event|periodicsync events=] are fired MUST be greater than or equal to this.
 
-A <dfn>time to fire</dfn> (a timestamp), which is the soonest time [=calculate a time to fire| calculated=] by the user agent at which it is appropriate to fire the next [=periodicSync event=] for the [=periodicsync registration=].
+A <dfn>time to fire</dfn> (a timestamp), which is the soonest time [=calculate a time to fire| calculated=] by the user agent at which it is appropriate to fire the next [=periodicsync event=] for the [=periodic sync registration=].
 
-A <dfn>count of retries</dfn> (a number), which is the number of retries attempted for the most recent [=periodicSync event=] for the [=periodicsync registration=].
+A <dfn>count of retries</dfn> (a number), which is the number of retries attempted for the most recent [=periodicsync event=] for the [=periodic sync registration=]. This is initially set to zero.
 
-A <dfn>registration state</dfn>, which is one of <dfn>pending</dfn>, <dfn>firing</dfn>, or <dfn>reregisteredWhileFiring</dfn>. It is initially set to [=pending=].
+A <dfn>state</dfn>, which is one of "<code>pending</code>", "<code>firing</code>", "<code>suspended</code>" or "<code>reregisteredWhileFiring</code>". It is initially set to "<code>pending</code>".
 </div>
 
+## Periodic Sync Scheduler ## {#periodic-sync-scheduler-construct}
```

Comment:
This would work, except when there are no registrations. The scheduler in this model would still be in a wait-then-check-for-registrations-and-process-them loop.

Reply:
Why's that a problem?

Reply:
Browser resources. The scheduler would be doing unnecessary work. Should we not care about that here?

Reply:
Specifications define behaviour, not performance optimisations.

You could add a note to say browsers may suspend and restart the loop to preserve battery, but it doesn't need to be normative because there's no observable behavioural difference.

Reply:
Ack. Done.